lib/Sema/MiscDiagnostics.h

```diff
@@ -64,6 +65,18 @@ bool diagnoseArgumentLabelError(TypeChecker &TC, const Expr *expr,
                                 bool isSubscript,
                                 InFlightDiagnostic *existingDiag = nullptr);
 
+/// If \c destExpr refers to a declaration with a non-owning attribute, such as
+/// 'weak' or 'unowned' and \c initializer refers to a class constructor,
+/// emit a warning that the assigned instance will be immediately deallocated.
+void diagnoseUnownedImmediateDeallocation(Expr *destExpr, Expr *initializer,
```

Comment:
Nitpick: More `const`s on unmodified AST node parameters, please!

Nitpick: Use `\p` for parameters in doc comments rather than `\c`.

(here and below)